jcl/src/java.base/share/classes/openj9/internal/management/ClassLoaderInfoBaseImpl.java

```diff
@@ -0,0 +1,38 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package openj9.internal.management;
+
+/**
+ * Container for classloader information.
+ */
+public class ClassLoaderInfoBaseImpl {
+	/**
+	 * @return the number of loaded classes
+	 */
+	public static native int getLoadedClassCountImpl();
```

Comment:
Why `int` here, but `long` below?

Reply:
This is inherited from existing code. 
https://github.com/eclipse/openj9/blob/0739a278d3a5cbc0c19e14e7fcb4d8164379b9ea/runtime/oti/j9nonbuilder.h#L765-L766
https://github.com/eclipse/openj9/blob/0739a278d3a5cbc0c19e14e7fcb4d8164379b9ea/runtime/jcl/common/mgmtclassloading.c#L45
https://github.com/eclipse/openj9/blob/0739a278d3a5cbc0c19e14e7fcb4d8164379b9ea/runtime/jcl/common/mgmtclassloading.c#L77
So this appears `long` is for `U_64` field values and `int` is to denote the diff of two `U_64`.

Reply:
This is good opportunity to fix that: `int` is not wide enough to capture the difference between two `U_64` values.

Reply:
As per javadoc `java.lang.management.ClassLoadingMXBean.getLoadedClassCount()` [1]:
```
int getLoadedClassCount()
Returns the number of classes that are currently loaded in the Java virtual machine.
Returns: the number of currently loaded classes.
```
On the other hand, `java.lang.management.ClassLoadingMXBean.getUnloadedClassCount()` has `long` as return value:
```
long getUnloadedClassCount()
Returns the total number of classes unloaded since the Java virtual machine has started execution.
Returns: the total number of unloaded classes.
```

So this is a public API specification, one returns `int` and the other returns `long`.
Current implementation is correct at returning an `int`. I think the expectation is that `int` is large enough to hold the number of classes currently loaded.

[1] https://docs.oracle.com/javase/8/docs/api/java/lang/management/ClassLoadingMXBean.html#getLoadedClassCount--
[2] https://docs.oracle.com/javase/8/docs/api/java/lang/management/ClassLoadingMXBean.html#getUnloadedClassCount--

Reply:
Note the internal api and jstat don't need to have the same restrictions as the getLoadedClassCount() public api.

Reply:
`com.ibm.java.lang.management.internal.ClassLoadingMXBeanImpl` implements `java.lang.management.ClassLoadingMXBean`, and its `public int getLoadedClassCount()` returns `ClassLoaderInfoBaseImpl.getLoadedClassCountImpl()` which should return `int` otherwise a `cast` to `int` is still required.

Reply:
Yes, a cast to `int` will be required, but we should consider deprecating that short-sighted API that returns `int`; applications are getting bigger all the time - 2 billion classes isn't unimaginable any more.

Reply:
Not disagreeing. Current casting is in native `Java_openj9_internal_management_ClassLoaderInfoBaseImpl_getLoadedClassCountImpl`, what's the benefit to move the cast to Java?
Can we hold such change until `java.lang.management. ClassLoadingMXBean` specification is ratified for `long` return value?

Reply:
Since you're changing the native method anyway, now seems like the right time to fix the return type. I'm fine if you want to defer adding new public API that would return `long` loaded class count. That future work will be a smaller change if we do this now.

Reply:
Sure, as you wish.

Running a personal build for sanity check.